src/webportal/src/app/job-submission/job-submission-page.jsx

```diff
@@ -0,0 +1,374 @@
+ * of this software and associated documentation files (the 'Software'), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import 'core-js/stable';
+import 'regenerator-runtime/runtime';
+import 'whatwg-fetch';
+
+import React, {useState, useCallback, useEffect, useMemo} from 'react';
+import {
+  Fabric,
+  Stack,
+  initializeIcons,
+  StackItem,
+} from 'office-ui-fabric-react';
+import {isNil, isEmpty} from 'lodash';
+import PropTypes from 'prop-types';
+
+import {JobInformation} from './components/job-information';
+import {getFormClassNames} from './components/form-style';
+import {SubmissionSection} from './components/submission-section';
+import {TaskRoles} from './components/task-roles';
+import Context from './components/context';
+import {listUserVirtualClusters} from './utils/conn';
+import {TaskRolesManager} from './utils/task-roles-manager';
+import {initTheme} from '../components/theme';
+
+// sidebar
+import {Parameters} from './components/sidebar/parameters';
+import {Secrets} from './components/sidebar/secrets';
+import {EnvVar} from './components/sidebar/env-var';
+import {DataComponent} from './components/data/data-component';
+import {ToolComponent} from './components/tools/tool-component';
+// models
+import {Topbar} from './components/topbar/topbar';
+import {JobBasicInfo} from './models/job-basic-info';
+import {JobTaskRole} from './models/job-task-role';
+import {JobData} from './models/data/job-data';
+import {JobProtocol} from './models/job-protocol';
+import {
+  getJobComponentsFromConfig,
+  isValidUpdatedTensorBoardExtras,
+} from './utils/utils';
+
+initTheme();
+initializeIcons();
+
+const formLayout = getFormClassNames().formLayout;
+
+const SIDEBAR_PARAM = 'param';
+const SIDEBAR_SECRET = 'secret';
+const SIDEBAR_ENVVAR = 'envvar';
+const SIDEBAR_DATA = 'data';
+const SIDEBAR_TOOL = 'tool';
+
+const loginUser = cookies.get('user');
+
+export const JobSubmissionPage = ({isSingle, setWizardStatus, yamlText}) => {
+  const [jobTaskRoles, setJobTaskRolesState] = useState([
+    new JobTaskRole({name: 'Default_Task_Role'}),
```

Comment:
k8s framework controller supports `[a-z0-9]{1,63}` only, better change `Default_Task_Role` to `taskrole`, etc

Reply:
done